source/Octopus.Versioning/Octopus/OctopusVersionParser.cs

```diff
@@ -1,10 +1,16 @@
 ﻿using System;
 using System.Text.RegularExpressions;
+using System.Linq;
 
 namespace Octopus.Versioning.Octopus
 {
     public class OctopusVersionParser
     {
+        /// <summary>
+        /// Versions can appear in URLs, and these characters are hard to work with in URLs, so we exclude them from any part of the version
+        /// </summary>
+        static readonly string[] IllegalChars = { "/", "%", "?", "#", "&" };
```

Comment:
I think I'd prefer to see us restrict to alpha-numerics, and only relax for specific characters; specifically `-` and maybe `_`.

Reply:
Maven allows for `A-Za-z0-9_\\-.`, so I've limited the chars to that.